src/main/java/org/folio/orders/utils/HelperUtils.java

```diff
@@ -70,10 +76,25 @@ public static JsonObject verifyAndExtractBody(Response response) {
     return response.getBody();
   }
 
+  /**
+   * If response http code is {@code 404}, empty {@link JsonObject} is returned. Otherwise verifies if response is successful and extracts body
+   * In case there was failed attempt to delete PO or particular PO line, the sub-objects might be already partially deleted.
+   * This check allows user to retry DELETE operation
+   *
+   * @param response response to verify
+   * @return empty {@link JsonObject} if response http code is {@code 404}, otherwise verifies if response is successful and extracts body
+   */
+  private static JsonObject verifyAndExtractBodyIfFound(Response response) {
+    if (response.getCode() == 404) {
+      return new JsonObject();
+    }
+    return verifyAndExtractBody(response);
+  }
+
   public static Adjustment calculateAdjustment(List<PoLine> lines) {
     Adjustment ret = null;
-    for (int i = 0; i < lines.size(); i++) {
-      Adjustment a = lines.get(i).getAdjustment();
+    for (PoLine line : lines) {
+      Adjustment a = line.getAdjustment();
```

Comment:
shouldn't we check for null here? a NPE be thrown if line is null?

Reply:
As far as I can see this method is called either from POST or GET order. Looking at the logic of those operations the PO line cannot be added to the list as null. So I guess there is no need to add this check.